networkit/matching.pyx

```diff
@@ -20,39 +20,144 @@ cdef class Matching:
 		return self
 
 	def match(self, node u, node v):
+		"""
+		Set two nodes u and v as each others matching partners.
+		
+		Parameters:
+		-----------
+		u : node
+		v : node
+		"""
 		self._this.match(u,v)
 
 	def unmatch(self, node u,  node v):
+		"""
+		Reset the two nodes u and v to unmatched.
+		
+		Parameters:
+		-----------
+		u : node
+		v : node
+		"""
 		self._this.unmatch(u, v)
 
 	def isMatched(self, node u):
+		"""
+		Check if node u is matched.
+		
+		Parameters:
+		-----------
+		u : node
+		
+		Returns:
+		-----------
+		bool
+			true if u has a matching partner.
+		"""
 		return self._this.isMatched(u)
 
 	def areMatched(self, node u, node v):
+		"""
+		Check if node u and v are matched to each other.
+		
+		Parameters:
+		-----------
+		u : node
+		v : node
+		
+		Returns:
+		-----------
+		bool
+			true if node u and v are matched to each other.
+		"""
 		return self._this.areMatched(u,v)
 
 	def isProper(self, Graph G):
+		"""
+		Check whether this is a proper matching in the graph, i.e. there is a edge between
+		each pair and if the matching partners of v is u then the matching partners of u is v.
+		
+		Parameters:
+		-----------
+		G : Graph
+			Graph to be checked.
+		Returns:
+		-----------
+		bool
+			true if this is a proper matching.
+		"""
 		return self._this.isProper(G._this)
 
 	def size(self, Graph G):
+		"""
+		Get the number of edges in this matching.
+		
+		Parameters:
+		-----------
+		G : Graph
+		
+		Returns:
+		-----------
+		count
+			total number of edges in the matching.
+		"""
 		return self._this.size(G._this)
 
 	def mate(self, node v):
+		"""
+		Get the matched neighbor of v if it exists, otherwise none.
```

Comment:
None is used in the C++ code, not sure that this holds also in Python.

Reply:
None also comes up in the python code. (see below with edge scores).

Reply:
What would the return value for a non-existent matching if not None?

Reply:
In `Globals.hpp` none is defined as `std::numeric_limits<index>::max()`, which is not the Python `None`. I think that, if there is no matching partner, this method just returns that value, and we should point this out in the docs. However, this behavior is not ideal, I think we should change the return value to `None` instead of +inf.

Reply:
Ok, I will change it to +inf for now. Maybe also open a new issue to change c++ to none to also be None in python?

Reply:
yes let's do that